addons/assigngear/functions/fnc_assignGear.sqf

```diff
@@ -180,23 +174,20 @@ _unit setUnitLoadout (configFile >> 'EmptyLoadout');
                 if !(_weapon isEqualTo '') then {_unit addWeapon _weapon};
             };
             case IDX_SIDEARM_MAGAZINE: { // sidearmmagazine
-                {
-                    if !(_x isEqualTo '') then {
-                        _loadedMagazines = handgunMagazine _unit;
-                        if (count _loadedMagazines > 0) then {
-                            {
-                                _unit removeHandgunItem _x;
-                                switch true do {
-                                    case (_unit canAddItemToVest _x): {_unit addItemToVest _x;};
-                                    case (_unit canAddItemToUniform _x): {_unit addItemToUniform _x;};
-                                    default {_unit addItemToBackpack _x;};
-                                };
-                            } forEach _loadedMagazines;
+                // Only attempt to unload magazines if the user specifies a sidearm magazine
+                if (count _x > 0) then {
```

Comment:
Can never be 0, as empty arrays are skipped: https://github.com/jdcaperon/TMF/blob/04679a279ac537f6c1da0d23831d0c60bcc18c40/addons/assigngear/functions/fnc_assignGear.sqf#L41

Reply:
I believe the `''` check is for the case where `primaryMagazine[] = {};`

Reply:
Actually that gets covered by line 41 too.

Reply:
Ah wait nah, it's for https://community.bistudio.com/wiki/selectRandom I think, when it returns Nothing.